Modules/_json.c

```diff
@@ -1549,14 +1570,25 @@ encoder_listencode_dict(PyEncoderObject *s, _PyUnicodeWriter *writer,
     if (_PyUnicodeWriter_WriteChar(writer, '{'))
         goto bail;
 
+    PyObject* newline_indent = 0;
+    PyObject* current_item_separator = s->item_separator;
     if (s->indent != Py_None) {
-        /* TODO: DOES NOT RUN */
         indent_level += 1;
-        /*
-            newline_indent = '\n' + (' ' * (_indent * _current_indent_level))
-            separator = _item_separator + newline_indent
-            buf += newline_indent
-        */
+        newline_indent = _create_newline_indent(s->indent, indent_level);
+        if (newline_indent == 0) {
+            goto bail;
+        }
+        current_item_separator = PyUnicode_Concat(current_item_separator, newline_indent);
+        if (current_item_separator == 0) {
+            Py_DECREF(newline_indent);
+            goto bail;
+        }
+
+        if (_PyUnicodeWriter_WriteStr(writer, newline_indent) < 0) {
+            Py_DECREF(current_item_separator);
+            Py_DECREF(newline_indent);
+            goto bail;
+        }
```

Comment:
I think we're leaking `newline_indent` here. I'd recommend running the json tests with `-R :` locally to check for refleaks.

Reply:
Check with -R is now passing